resources/assets/lib/utils/beatmapset-helper.ts

```diff
@@ -15,3 +22,32 @@ export function nominationsCount(nominations: BeatmapsetNominationsInterface, ty
 export function showVisual(beatmapset: BeatmapsetJson) {
   return !beatmapset.nsfw || (currentUser?.user_preferences?.beatmapset_show_nsfw ?? false);
 }
+
+export function toggleFavourite(beatmapset: BeatmapsetJson) {
+  const retryCallback = () => toggleFavourite(beatmapset);
+
+  if (core.userLogin.showIfGuest(retryCallback)) return;
+
+  const add = !beatmapset.has_favourited;
+
+  // fake immediate change
+  beatmapset.has_favourited = add;
+  beatmapset.favourite_count += add ? 1 : -1;
+
+  $.ajax(route('beatmapsets.favourites.store', { beatmapset: beatmapset.id }), {
```

Comment:
should be prefixed with `void` to mark the promise as being unhandled apparently https://github.com/typescript-eslint/typescript-eslint/blob/v4.17.0/packages/eslint-plugin/docs/rules/no-floating-promises.md